src/hal/src/format.rs

```diff
@@ -148,24 +148,27 @@ bitflags!(
     pub struct ImageFeature: u32 {
         /// Image view can be sampled.
         const SAMPLED = 0x1;
-        /// Image view can be used as storage image.
-        const STORAGE = 0x2;
-        /// Image view can be used as storage image (with atomics).
-        const STORAGE_ATOMIC = 0x4;
+        /// Image can be sampled with a (mipmap) linear sampler or as blit source with linear sampling.
+        const SAMPLED_LINEAR = 0x10000 | ImageFeature::SAMPLED.bits() | ImageFeature::BLIT_SRC.bits();
+
+        /// Image view can be used as storage image with exclusive read & write access.
+        const STORAGE = 0x10;
+        /// Image view can be used as storage image with simultaneous read/write access.
+        const STORAGE_READ_WRITE = 0x20 | ImageFeature::STORAGE.bits();
+        /// Image view can be used as storage image with atomics.
+        const STORAGE_ATOMIC = 0x40 | ImageFeature::STORAGE_READ_WRITE.bits();
```

Comment:
We should probably not do that. More specifically, having the constant flags that are neither combinations of existing constants, or single-bit flags. It becomes a bit confusing, and there isn't a strong reason to go this route.

Interestingly, this looks more like an enum to me:
```rust
enum StorageSupport {
  Basic,
  ReadWrite,
  Atomic,
}
```
Of course I'm not suggesting you to make one, just thought about it as an interesting direction in the future.

Reply:
I'll remove the combinations then 🤷 
Agreed on the enum, that would be cleaner. But yeah let's leave that for another time as this would ripple out much more then.